### DISCUSSION TITLE
e2e-testing-mfa-environment-playwright-auth-session/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#626] [Comments]
# e2e-testing-mfa-environment-playwright-auth-session/

Easily perform E2E testing in MFA environments using Playwright's authenticated session state. Learn how to automate your tests without logging in every time.

https://www.eliostruyf.com/e2e-testing-mfa-environment-playwright-auth-session/?utm_source=collab365&utm_medium=collab365today&utm_campaign=daily_digest

**Comment by MartinLaplante** [comment]
The FedAuth cookie alone will work if you only need access to one site collection, but if you need a bigger scope do you also store the rtFA cookie?

**Reply by estruyf** [reply]
The FedAuth cookie was taken as an example. All cookies are stored and the approach works towards Microsoft Teams and Power Apps too.
